pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -595,6 +642,12 @@ private LLCRealtimeSegmentZKMetadata updateCommittingSegmentZKMetadata(String re
     committingSegmentZKMetadata.setPartitionMetadata(getPartitionMetadataFromSegmentMetadata(segmentMetadata));
 
     persistSegmentZKMetadata(realtimeTableName, committingSegmentZKMetadata, stat.getVersion());
+
+    // cache peer download segment for fix
```

Comment:
Change to "Add a segment not successfully uploaded to deep store to a queue managed by Pinot controller for upload retry later."